message_ix_models/model/material/build.py

```diff
@@ -233,3 +203,57 @@ def get_spec() -> Mapping[str, ScenarioInfo]:
             remove.set[set_name].extend(config.get("remove", []))
 
     return dict(require=require, add=add, remove=remove)
+
+
+def make_spec(regions: str, materials: str or None = SPEC_LIST) -> Spec:
+    sets: Dict[str, Any] = dict()
+    materials = ["common"] if not materials else materials
+    # Overrides specific to regional versions
+    tmp = dict()
+    # technology.yaml currently not used in Materials
+    for fn in ["set.yaml"]:  # , "technology.yaml"):
+        # Field name
+        name = fn.split(".yaml")[0]
+
+        # Load and store the data from the YAML file: either in a subdirectory for
+        # context.model.regions, or the top-level data directory
+        path = path_fallback(regions, fn).relative_to(package_data_path())
+        # tmp[name] = load_private_data(*path.parts)
+        tmp[name] = load_package_data(*path.parts)
+
+    # Merge contents of technology.yaml into set.yaml
+    # technology.yaml currently not used in Materials
+    sets.update(tmp.pop("set"))
+
+    s = Spec()
+
+    # Convert some values to codes
+    for material in materials:
+        for set_name in sets[material]:
+            if not all(
+                [
+                    isinstance(item, list)
+                    for sublist in sets[material][set_name].values()
+                    for item in sublist
+                ]
+            ):
+                generate_set_elements(sets[material], set_name)
+
+            # Elements to add, remove, and require
+            for action in {"add", "remove", "require"}:
+                s[action].set[set_name].extend(sets[material][set_name].get(action, []))
+            try:
+                s.add.set[f"{set_name} indexers"] = sets[material][set_name]["indexers"]
+            except KeyError:
+                pass
+
+    # The set of required nodes varies according to context.model.regions
+    codelist = regions
+    try:
+        s["require"].set["node"].extend(map(str, get_region_codes(codelist)))
+    except FileNotFoundError:
+        raise ValueError(
+            f"Cannot get spec for MESSAGEix-Materials with regions={codelist!r}"
+        ) from None
```

Comment:
Testing these two lines (254 and 255) might not be strictly necessary, but if you wanted to do it, I'd recommend making these lines their own function, that does just one job: take a `Spec` object and extend the `node` set. This would likely be easier to test than this whole `make_spec()` function as it requires less setup.